tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -2379,6 +2380,33 @@ public function testDecimal(): void
         $this->assertValidationMessage($fieldName, $rule, $expectedMessage, $places);
     }
 
+    /**
+     * Tests the localized decimal proxy method
+     */
+    public function testLocalizedDecimal(): void
+    {
+        $validator = new Validator();
+        $this->assertProxyMethod($validator, 'localizedDecimal', 2, [2]);
+        $this->assertNotEmpty($validator->validate(['username' => 10.1]));
+
+        $fieldName = 'field_name';
+        $rule = 'decimal';
+        $expectedMessage = 'The provided value must be decimal with `2` decimal places';
+        $places = 2;
+        $this->assertValidationMessage($fieldName, $rule, $expectedMessage, $places);
+
+        $expectedMessage = 'The provided value must be decimal with any number of decimal places, including none';
+        $places = null;
+        $this->assertValidationMessage($fieldName, $rule, $expectedMessage, $places);
+
+        $this->skipIf(DS === '\\', 'The locale is not supported in Windows and affects other tests.');
+        $this->skipIf(Locale::setDefault('pl_PL') === false, "The Polish locale isn't available.");
+
+        $this->assertEmpty($validator->validate(['username' => '1 200,99']));
+
+        Locale::setDefault(Validation::DEFAULT_LOCALE);
+    }
```

Comment:
Do we have a test for duplicate thousands or decimal symbol?

Reply:
Can you give examples? I can add them.

Reply:
I was looking at the locale-specific string replace and it should prevent validation of something like:
```
123,,456..000
```

Reply:
Assuming `en_US` locale `123,,456.0` will pass since the thousand separators are removed before the regex validation. `123,456..0` with duplicate decimal symbol will fail.

Reply:
I have updated the code so that value like `123,,456.0` also fails validation.